localstack-core/localstack/services/events/archive.py

```diff
@@ -0,0 +1,167 @@
+import json
+import logging
+from datetime import datetime, timezone
+
+from botocore.client import BaseClient
+
+from localstack.aws.api.events import (
+    ArchiveState,
+    Arn,
+    TargetId,
+    Timestamp,
+)
+from localstack.aws.connect import connect_to
+from localstack.services.events.models import (
+    Archive,
+    ArchiveDescription,
+    ArchiveName,
+    EventPattern,
+    FormattedEvent,
+    FormattedEventList,
+    RetentionDays,
+    RuleName,
+)
+from localstack.services.events.utils import extract_event_bus_name
+from localstack.utils.aws.client_types import ServicePrincipal
+
+LOG = logging.getLogger(__name__)
+
+
+class ArchiveService:
+    def __init__(
+        self,
+        archive_name: ArchiveName,
+        region: str,
+        account_id: str,
+        event_source_arn: Arn,
+        description: ArchiveDescription,
+        event_pattern: EventPattern,
+        retention_days: RetentionDays,
+    ):
+        self.archive = Archive(
+            archive_name,
+            region,
+            account_id,
+            event_source_arn,
+            description,
+            event_pattern,
+            retention_days,
+        )
+        self.set_state(ArchiveState.CREATING)
+        self.set_creation_time()
+        self.client: BaseClient = self._initialize_client()
+        self.event_bus_name = extract_event_bus_name(event_source_arn)
+
+        self.rule_name = self._create_archive_rule()
+        self.target_id = self._create_archive_target()
+        self.set_state(ArchiveState.ENABLED)
+
+    def __getattr__(self, name):
+        return getattr(self.archive, name)
+
+    @property
+    def archive_name(self) -> ArchiveName:
+        return self.archive.name
+
+    @property
+    def archive_arn(self) -> Arn:
```

Comment:
makes it easier to translate from external variable name archive_arn to internal variable name arn